lib/node_modules/@stdlib/math/base/special/roundsd/test/test.native.js

```diff
@@ -0,0 +1,257 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var PI = require( '@stdlib/constants/float64/pi' );
+var PINF = require( '@stdlib/constants/float64/pinf' );
+var NINF = require( '@stdlib/constants/float64/ninf' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var isNegativeZero = require( '@stdlib/math/base/assert/is-negative-zero' );
+var isPositiveZero = require( '@stdlib/math/base/assert/is-positive-zero' );
+var roundsd = require( './../lib' );
+
+
+// FIXTURES //
+
+var base2 = require( './fixtures/julia/base_2_sigfigs_4.json' );
+var base16 = require( './fixtures/julia/base_16_sigfigs_4.json' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof roundsd, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided `NaN`', function test( t ) {
+	var v;
+
+	v = roundsd( NaN, 2 );
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
+
+	v = roundsd( 12368.0, NaN );
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
+
+	v = roundsd( NaN, NaN );
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
+
+	v = roundsd( NaN, NaN, 10 );
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
+
+	v = roundsd( NaN, 2, 10 );
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
+
+	v = roundsd( 3.14, NaN, 10 );
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
+
+	v = roundsd( 3.14, 2, NaN );
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
+
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided `n = +-infinity`', function test( t ) {
+	var v;
+
+	v = roundsd( PI, PINF );
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
+
+	v = roundsd( PI, NINF );
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
+
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided `n < 1`', function test( t ) {
+	var v;
+
+	v = roundsd( PI, 0 );
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
+
+	v = roundsd( PI, -1 );
+	t.strictEqual( isnan( v ), true, 'returns NaN' );
```

Comment:
```suggestion
	v = roundsd( PI, 0, 10 );
	t.strictEqual( isnan( v ), true, 'returns NaN' );

	v = roundsd( PI, -1, 10 );
	t.strictEqual( isnan( v ), true, 'returns NaN' );
```

All of your tests should be refactored to have 3 arguments for `roundsd`.